de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java

```diff
@@ -979,51 +993,67 @@ public void removeActivityConsumer(IActivityConsumer consumer) {
 
     @Override
     public boolean isShared(IResource resource) {
-        return projectMapper.isShared(resource);
+        return sharedReferencePointMapper.isShared(resource);
     }
 
     @Override
     public List<IResource> getSharedResources() {
-        return projectMapper.getPartiallySharedResources();
+
+        return sharedReferencePointMapper.getPartiallySharedResources();
     }
 
     @Override
     public String getProjectID(IProject project) {
-        return projectMapper.getID(project);
+        return sharedReferencePointMapper.getID(project.getReferencePoint());
     }
 
     @Override
     public IProject getProject(String projectID) {
-        return projectMapper.getProject(projectID);
+        return referencePointManager.get(sharedReferencePointMapper
+            .getReferencePoint(projectID));
     }
 
     @Override
     public Map<IProject, List<IResource>> getProjectResourcesMapping() {
-        return projectMapper.getProjectResourceMapping();
+        Map<IReferencePoint, List<IResource>> referencePointResourceMap = sharedReferencePointMapper
+            .getReferencePointResourceMapping();
+        Map<IProject, List<IResource>> projectResourceMap = new HashMap<IProject, List<IResource>>();
+
+        for (IReferencePoint referencePoint : referencePointResourceMap
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Iterate over the "entrySet" instead of the "keySet". ([squid:S2864](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS2864))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=257&issues=AWbeHvOCSgN4uCclRsG4&open=AWbeHvOCSgN4uCclRsG4)

[ISSUE_KEY]: AWbeHvOCSgN4uCclRsG4